src/native/corehost/bundle/header.h

```diff
@@ -87,11 +89,13 @@ namespace bundle
         const location_t& runtimeconfig_json_location() const { return m_v2_header.runtimeconfig_json_location; }
         bool is_netcoreapp3_compat_mode() const { return m_v2_header.is_netcoreapp3_compat_mode(); }
 
-        static const uint32_t major_version = 2;
-        static const uint32_t minor_version = 0;
+        const uint32_t major_version() const { return m_major_version; };
+        const uint32_t minor_version() const { return m_minor_version; };
 
     private:
         int32_t m_num_embedded_files;
+        uint32_t m_major_version;
+        uint32_t m_minor_version;
```

Comment:
This is a bit tricky. Currently we fail on anything but 6.0. But in theory we should support 2.0 as well. Scenario is: running 5.0 single-file framework dependent application on 6.0 framework (roll-forward).

We should at least file an issue for this.

Reply:
As-is there's no reason to store the versions. But if we want to support the 5.0 on 6.0, then we will probably need to store these.

Reply:
I will add versioning for the header and a test.